nixos/modules/programs/kbdlight.nix

```diff
@@ -11,6 +11,13 @@ in
 
   config = mkIf cfg.enable {
     environment.systemPackages = [ pkgs.kbdlight ];
-    security.setuidPrograms = [ "kbdlight" ];
+
+    security.permissionsWrappers.setuid =
+    [ { program = "kbdlight";
+        source  = "${pkgs.kbdlight.out}/bin/kbdlight";
+        owner   = "root";
```

Comment:
Thanks, @edolstra made the same comment and suggested an attrset (instead of a list of attrsets) with sane defaults. I'm going to give that a go when I have some free-time.